camera: refactor camera organisation

lahm86 opened the pull request:
Resolves #1654.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

This moves camera functions to module scope where applicable, introduces/standardizes `const` usage throughout the module, and reorders function declarations and implementations. No change in functionality.

lahm86 committed:
camera: move functions to module scope

lahm86 committed:
camera: standardize const usage

rr- reviewed (commented):
Nitpick – currently we mix static and non-static functions; could we make sure all the module-scoped (eg static) functions are placed before public API?

lahm86 committed:
camera: reorder function declarations

This reorders camera function declarations such that module functions
are first, and it tries to keep related functions closer together.

lahm86 commented:
> Nitpick – currently we mix static and non-static functions; could we make sure all the module-scoped (eg static) functions are placed before public API?

Done. I've tidied up the general order too to try to keep related things together, and match with the header declaration order. Should be somewhat easier to navigate.

rr- reviewed (approved):
Thanks!
